integration-cli/cli/build/fakestorage/storage.go

```diff
@@ -130,6 +135,15 @@ func newRemoteFileServer(t testingT, ctx *fakecontext.Fake) *remoteFileServer {
 
 	ensureHTTPServerImage(t)
 
+	dockerHostURL, err := url.Parse(request.DaemonHost())
+	if err != nil {
+		t.Fatalf("unable to parse daemon host URL: %v", err)
+	}
+
+	if strings.HasPrefix(dockerHostURL.String(), "unix:///") {
+		t.Skip(fmt.Sprintf("e2e run : daemon is remote but docker host points to a unix socket"))
```

Comment:
Because when running the `e2e` image, it is set as `RemoteDaemon` (because we're in a container, we don't have access to the host filesystem), but the host *can be* `unix:///var/run/…` too.

Honestly, I think, in a near future, the e2e image shouldn't set `RemoteDaemon` but sthg else that would allow us to know that we are in this special case.

Reply:
I'm still not understanding. 

This code is used for both e2e and CI for PRs. Isn't this assuming it's always remote ? Where's the check for `IsRemoveHost` ?

Reply:
We discussed in slack, I think this could be moved to `New()` where the local vs remote is checked. It could be a switch/case with 3 cases: local, remote, "remote but doesn't use a tcp port for host"